File: docs-examples/example-java/src/test/java/io/micronaut/rabbitmq/docs/consumer/connection/ConnectionSpec.java
package io.micronaut.rabbitmq.docs.consumer.connection;

import java.net.URI;
import java.util.HashMap;
import java.util.Map;

import io.micronaut.context.annotation.Property;
import io.micronaut.rabbitmq.testcontainers.RabbitMQ;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import io.micronaut.test.support.TestPropertyProvider;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;

import static java.util.concurrent.TimeUnit.SECONDS;
import static org.awaitility.Awaitility.await;

@MicronautTest
@Property(name = "spec.name", value = "ConnectionSpec")
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
class ConnectionSpec implements TestPropertyProvider {

    @Test
    void testProductClientAndListener(ProductClient productClient, ProductListener productListener) {

// tag::producer[]
productClient.send("connection-test".getBytes());
// end::producer[]

        await().atMost(10, SECONDS).until(() ->
                productListener.messageLengths.size() == 1 &&
                productListener.messageLengths.get(0).equals("connection-test")
        );
    }

    @Override
    public Map<String, String> getProperties() {
        Map<String, String> m = new HashMap<>(RabbitMQ.getProperties());
        String uri = m.get("rabbitmq.uri");
        String port = String.valueOf(URI.create(uri).getPort());
        m.put("rabbitmq.servers.product-cluster.port", port);
        return m;
    }
}

File: docs-examples/example-python/src/test/python/micronaut/rabbitmq/docs/consumer/connection/ConnectionSpec.py
from typing import Annotated

from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.rabbitmq.docs.Await import Await
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from .ProductClient import ProductClient
from .ProductListener import ProductListener


# The "rabbitmq-cluster" environment configures the test container as the "product-cluster" server
# (see the Java RabbitMQTestConfigurer of this project)
@MicronautTest(environments=["rabbitmq", "rabbitmq-cluster"])
@Property(name="spec.name", value="ConnectionSpec")
class ConnectionSpec:
    product_client: Annotated[ProductClient, Inject]
    product_listener: Annotated[ProductListener, Inject]

    @Test
    def test_product_client_and_listener(self) -> None:
# tag::producer[]
        self.product_client.send(b"connection-test")
# end::producer[]

        Await.until(lambda: self.product_listener.message_lengths == ["connection-test"], timeout=10)
